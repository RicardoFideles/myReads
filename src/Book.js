import React, { Component } from 'react'
import If from './If'

class Book extends Component {
    render() {
        const {book, onUpdateBook, shelves}  = this.props
        const authors = book.authors !== undefined ? book.authors : []
        return (
                <div className="book">
                    <div className="book-top">
                        <div className="book-cover" style={{ width: 128, height: 193, backgroundImage: `url("${book.imageLinks.thumbnail}")` }}></div>
                        <div className="book-shelf-changer">
                            <select onChange={(event) => onUpdateBook(event.target.value, book)} value={book.shelf !== undefined ? book.shelf : '' }> 
                                <option value="" disabled>Move to...</option>
                                {
                                    shelves.map(shelf => (
                                        <option key={shelf.id} value={shelf.id}>{shelf.title}</option>
                                    ))
                                }
                                <If test={book.shelf !== undefined}>
                                    <option value="none">None</option>
                                </If>
                            </select>
                        </div>
                    </div>
                    <div className="book-title">{book.title}</div>
                    {authors.map(author => (
                        <div key={author} className="book-authors">{author}</div>
                    ))}
                    
                </div>
            )
    }
}

export default Book;