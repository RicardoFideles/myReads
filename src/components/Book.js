import React from 'react'
import { Link } from 'react-router-dom'
import BookSelfChanger from './BookSelfChanger'

const Book = props => {
    const {book, onUpdateBook, shelves, currentBooks}  = props
    const authors = book.authors !== undefined ? book.authors : []
    return (
        <div className="book">
            <div className="book-top">
                <Link to={`/book/${book.id}`}>
                    <div className="book-cover" style={{ width: 128, height: 193, backgroundImage: `url("${book.imageLinks.thumbnail}")` }}></div>
                </Link>
                <BookSelfChanger currentBooks={currentBooks} onUpdateBook={onUpdateBook} book={book} shelves={shelves}/>
            </div>
            <div className="book-title">{book.title}</div>
            {authors.map(author => (
                <div key={author} className="book-authors">{author}</div>
            ))}
        </div>
    )
}

export default Book