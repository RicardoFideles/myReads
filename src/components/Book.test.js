import React from "react"
import Book from "./Book"
import {shallow} from "enzyme"
import Adapter from 'enzyme-adapter-react-15'
import * as Enzyme from "enzyme"

Enzyme.configure({ adapter: new Adapter() })

it('renders without crashing', () => {
    const onUpdateBook = jest.fn()
    const shelves = []
    const currentBooks = []
    const book = { id : 1, imageLinks : { thumbnail : "test"}, title : "Jest In Action", authors : [ "Ricardo", "Nathaly"] }
    expect(shallow(<Book shelves={shelves} currentBooks={currentBooks} book={book} onUpdateBook={onUpdateBook}/>)).toMatchSnapshot()
})
