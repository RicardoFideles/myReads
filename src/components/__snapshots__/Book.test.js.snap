// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="book"
>
  <div
    className="book-top"
  >
    <Link
      replace={false}
      to="/book/1"
    >
      <img
        className="book-cover"
        src="test"
      />
    </Link>
    <BookSelfChanger
      book={
        Object {
          "authors": Array [
            "Ricardo",
            "Nathaly",
          ],
          "id": 1,
          "imageLinks": Object {
            "thumbnail": "test",
          },
          "title": "Jest In Action",
        }
      }
      currentBooks={Array []}
      onUpdateBook={[Function]}
      shelves={Array []}
    />
  </div>
  <div
    className="book-title"
  >
    Jest In Action
  </div>
  <div
    className="book-authors"
    key="Ricardo"
  >
    Ricardo
  </div>
  <div
    className="book-authors"
    key="Nathaly"
  >
    Nathaly
  </div>
</div>
`;
